runners/google-cloud-dataflow-java/pom.xml

```diff
@@ -122,6 +122,50 @@
         </plugins>
       </build>
     </profile>
+
+    <!-- A profile that adds an integration test phase for streaming if and only if
+         the streamingValidatesRunnerPipelineOptions maven property has been set.
+         It should be set to a valid PipelineOptions JSON string. -->
+    <profile>
```

Comment:
Obviously having a base set of pipeline options and hacking it (the JSON?) a bit in each execution was my first thought. Off the top of my head, I don't know a good way to do it cleanly. I can think of hacks (like extra side channels). Do you have suggestions?